pkg/infrastructure/platform/platform.go

```diff
@@ -95,7 +94,7 @@ func ProviderForPlatform(platform string, fg featuregates.FeatureGate) (infrastr
 		if types.ClusterAPIFeatureGateEnabled(platform, fg) {
 			return clusterapi.InitializeProvider(vspherecapi.Provider{}), nil
 		}
-		return terraform.InitializeProvider(vsphere.PlatformStages), nil
+		return nil, fmt.Errorf("installing on vSphere via terraform is no longer supported")
```

Comment:
Now that capi/vsphere is the default, we could just drop this error and remove the `types.ClusterAPIFeatureGateEnabled` check. We can do that as a follow-up.